lib/logstash/codecs/auto_flush.rb

```diff
@@ -11,16 +11,21 @@ def start
     # can't start if pipeline is stopping
     return self if stopped?
     if pending?
-      @task.reset
+      @task.cancel
```

Comment:
trying to wrap around my head on this one.

I think the second example is better:

just dumping my brain here.

1. We check if the the task is pending. we try to cancel it, return true if successful, on false it will prevent any other run
2. Check if the task is finished, create a new task if its the case, and exit
3. block on the value of execution, `@task.value`

Any specific reason why the last line return self and not the other return?
https://github.com/logstash-plugins/logstash-codec-multiline/pull/24/files#diff-4ffb517712c74a5e9ee0fcf4800b6f0cR20